clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/cache/client/AbstractCacheClient.java

```diff
@@ -0,0 +1,63 @@
+package com.netflix.spinnaker.clouddriver.ecs.cache.client;
+
+import com.netflix.spinnaker.cats.cache.Cache;
+import com.netflix.spinnaker.cats.cache.CacheData;
+import com.netflix.spinnaker.clouddriver.ecs.cache.Keys;
+
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Set;
+
+public abstract class AbstractCacheClient<T> {
+
+  private final String keyNamespace;
+  private final Cache cacheView;
+
+  AbstractCacheClient(Cache cacheView, String keyNamespace) {
+    this.cacheView = cacheView;
+    this.keyNamespace = keyNamespace;
+  }
+
+  protected abstract T convert(CacheData cacheData);
+
+  public Collection<T> getAll() {
+    Collection<CacheData> allData = cacheView.getAll(keyNamespace);
+    return convertAll(allData);
+  }
+
+  public Collection<T> getAll(String account, String region) {
+    Collection<CacheData> data = fetchFromCache(account, region);
+    return convertAll(data);
+  }
+
+  public T get(String key) {
+    return convert(cacheView.get(keyNamespace, key));
+  }
+
+  private Collection<T> convertAll(Collection<CacheData> cacheData) {
+    Set<T> itemSet = new HashSet<>();
+    for (CacheData cacheDatum : cacheData) {
+      itemSet.add(convert(cacheDatum));
+    }
+    return itemSet;
+  }
+
+  /**
+   * @param account name of the AWS account, as defined in clouddriver.yml
+   * @param region  is not used in AWS as IAM is region-agnostic
```

Comment:
The `region` parameter description is copy-pasted wrong :)

Reply:
Done.